modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportZFIN.pm

```diff
@@ -0,0 +1,206 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+
+=head1 CONTACT
+
+ Please email comments or questions to the public Ensembl
+ developers list at <http://lists.ensembl.org/mailman/listinfo/dev>.
+
+ Questions may also be sent to the Ensembl help desk at
+ <http://www.ensembl.org/Help/Contact>.
+
+=cut
+
+
+=head1 ImportZFIN
+
+This module imports ZFIN (The Zebrafish Information Network) data.
+
+=cut
+
+package Bio::EnsEMBL::Variation::Pipeline::PhenotypeAnnotation::ImportZFIN;
+
+use strict;
+use warnings;
+
+use File::Path qw(make_path);
+use File::stat;
+use POSIX 'strftime';
+use LWP::Simple;
+
+use base ('Bio::EnsEMBL::Variation::Pipeline::PhenotypeAnnotation::BasePhenotypeAnnotation');
+
+my %source_info;
+
+sub fetch_input {
+  #create output folder structure and fetches input files
+  my $self = shift;
+
+  my $pipeline_dir = $self->required_param('pipeline_dir');
+  my $species      = $self->required_param('species');
+
+  $self->debug($self->param('debug_mode'));
+  $self->core_db_adaptor($self->get_species_adaptor('core'));
+  $self->variation_db_adaptor($self->get_species_adaptor('variation'));
+
+  # import specific constants
+  %source_info = (source_description => 'The zebrafish model organism database',
+                  source_url => 'https://www.zfin.org/',
+                  object_type => 'Gene',
+                  #source_version  will be set based on the date of the fetched input file  (year/month/day-> yyyymmdd)
+
+                  source_status => 'germline',
+
+                  source_name => 'ZFIN',        #source name in the variation db
+                  source_name_short => 'ZFIN',  #source identifier in the pipeline
+                  );
+  my $zfin_url = 'https://zfin.org/downloads/phenoGeneCleanData_fish.txt';
+  my $inputFile = 'phenoGeneCleanData_fish.txt';
```

Comment:
Looks like this variable could be used to build the URL just above instead of duplication of the filename